docs/specification.md

```diff
@@ -56,2103 +56,3622 @@ This document provides the detailed technical specification for the A2A protocol
+#### 4.1.9. Artifact
 
-<div class="grid cards" markdown>
+Represents a tangible output generated by the agent during a task.
 
-=== "JSON-RPC"
-    -   **URL:** `tasks/cancel`
-    -   **HTTP Method:** `POST`
-    -   **Payload**: [`TaskIdParams`](#751-taskidparams-object-for-taskscancel-and-taskspushnotificationconfigget)
-    -   **Response**: [`Task`](#61-task-object)
+```proto
+--8<-- "specification/grpc/a2a.proto:Artifact"
+```
 
-=== "gRPC"
-    -   **URL:** `CancelTask`
-    -   **HTTP Method:** `POST`
-    -   **Payload:**
-        ```proto
-        message CancelTaskRequest{
-          // name=tasks/{id}
-          string name;
-        }
-        ```
-    -   **Response**: [`Task`](#61-task-object)
-
-=== "REST"
-    -   **URL:** `/v1/tasks/{id}:cancel`
-    -   **HTTP Method:** `POST`
-    -   **Payload:**
-        ```typescript
-        {
-          name: string
-        }
-        ```
-    -   **Response**: [`Task`](#61-task-object)
+**Fields:**
 
-</div>
+- **`artifactId`** (required, string): Unique identifier (e.g. UUID) for the artifact. It must be at least unique within a task.
+- **`name`** (optional, string): A human readable name for the artifact.
+- **`description`** (optional, string): A human readable description of the artifact.
+- **`parts`** (required, array of [`Part`](#416-part)): The content of the artifact. Must contain at least one part.
+- **`metadata`** (optional, object): Optional metadata included with the artifact.
+- **`extensions`** (optional, array of strings): The URIs of extensions that are present or contributed to this Artifact.
 
-- **Response `error` type (on failure)**: [`JSONRPCError`](#612-jsonrpcerror-object) (e.g., [`TaskNotFoundError`](#82-a2a-specific-errors), [`TaskNotCancelableError`](#82-a2a-specific-errors)).
+### 4.2. Streaming Events
 
-#### 7.5.1. `TaskIdParams` Object (for `tasks/cancel` and `tasks/pushNotificationConfig/get`)
+#### 4.2.1. TaskStatusUpdateEvent
+<span id="4192-taskstatusupdateevent"></span><span id="722-taskstatusupdateevent-object"></span>
 
-A simple object containing just the task ID and optional metadata.
+Carries information about a change in task status during streaming.
 
-{{ render_spec_tabs('TaskIdParams') }}
+```proto
+--8<-- "specification/grpc/a2a.proto:TaskStatusUpdateEvent"
+```
 
-### 7.6. `tasks/pushNotificationConfig/set`
+**Fields:**
 
-Sets or updates the push notification configuration for a specified task. This allows the client to tell the server where and how to send asynchronous updates for the task. Requires the server to have `AgentCard.capabilities.pushNotifications: true`.
+- **`taskId`** (required, string): The id of the task that is changed.
+- **`contextId`** (required, string): The id of the context that the task belongs to.
+- **`status`** (required, [`TaskStatus`](#412-taskstatus)): The new status of the task.
+- **`final`** (required, boolean): Whether this is the last status update expected for this task.
```

Comment:
will this be removed? per https://github.com/a2aproject/A2A/pull/1160/files?diff=split&w=0#r2517872178